### PR TITLE
Fix "make clean"

### PR DESCRIPTION
**Pull request by MetRonnie**
Added new paths to auto-generated content introduced by #212

https://github.com/cylc/cylc-doc/blob/8088b4f2c8a603744b7b0c444e79f69c0ac6885e/.gitignore#L24-L26

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,6 +19,9 @@ help:
 clean:
 	@$(SPHINXBUILD) -M clean "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
 	# remove auto-generated content
+	rm -rf src/plugins/main-loop/built-in
+	rm -rf src/user-guide/task-implementation/job-runner-handlers
+	# remove auto-generated content (old)
 	rm -rf src/user-guide/plugins/main-loop/built-in
```
Comment (oliver-sanders) [inline Makefile:25]:
```suggestion
```

File: Makefile [hunk 2]
```diff
@@ -19,6 +19,9 @@ help:
...
 	rm -rf src/user-guide/batch-sys-handlers
 	rm -rf src/user-guide/job-runner-handlers
```
Comment (oliver-sanders) [inline Makefile:27]:
```suggestion
```

Comment (MetRonnie) [reply Makefile:27]: The idea was to remove old files that might be kicking around due to checking out between different branches, is that not needed?

Comment (oliver-sanders) [reply Makefile:27]: Can do with `git clean` rather than leaving old paths in a makefile.

**Review by oliver-sanders — APPROVED** [review]

**Pull request merged**